[pcl] Qhull is disabled in pcl_config.h, causes pcl::ConvexHull to not compile or (with workaround) to link

**Host Environment**
- OS: Windows 10 Pro 1909
- Compiler: v142 (Visual Studio 2019)

**To Reproduce**

1. `.\vcpkg install pcl:x64-windows`
2. Examine `.\vcpkg\installed\x64-windows\include\pcl\pcl_config.h` for `/* #undef HAVE_QHULL */`. Expected: `#define HAVE_QHULL`


**Additional context**
Using yesterday's Revision 95836f4883ffb481b7dd5f89569bc6ef55c31a7e of Vcpkg the problem occurs. Two weeks ago it worked fine with 807a7987612bbd702be9d9b8069124be62aff857. In the meantime issue https://github.com/microsoft/vcpkg/issues/13881, that addressed a related problem, was resolved and PR https://github.com/microsoft/vcpkg/pull/17437 was merged to master. Maybe a new bug was introduced?

`.\vcpkg\installed\x64-windows\bin\pcl_surface.dll` does not export any `pcl::ConvexHull` symbols or anything with `hull`.

Creating a test project with the environment set to the `.\vcpkg\installed\x64-windows\include` and `.\vcpkg\installed\x64-windows\lib` directories, linking all *.lib files in the latter.
Naively using pcl::ConvexHull gives compiler errors:

```
// ConvexHull Test.cpp

#include "pcl/point_types.h"
#include "pcl/point_cloud.h"
#include "pcl/surface/convex_hull.h"

int main() {
    pcl::PointCloud<pcl::PointXYZ> pc;
    pcl::ConvexHull<pcl::PointXYZ> hull;

    hull.reconstruct(pc);
}
```
Build log:
```
1>C:\ConvexHull Test\ConvexHull Test.cpp(12,10): error C2039: 'ConvexHull': is not a member of 'pcl'
1>C:\ConvexHull Test\ConvexHull Test.cpp(12,20): error C2065: 'ConvexHull': undeclared identifier
1>C:\ConvexHull Test\ConvexHull Test.cpp(12,26): error C2275: 'pcl::PointXYZ': illegal use of this type as an expression
1>C:\ConvexHull Test\ConvexHull Test.cpp(12,36): error C2065: 'hull': undeclared identifier
1>C:\ConvexHull Test\ConvexHull Test.cpp(14,5): error C2065: 'hull': undeclared identifier
```

**Workaround** (kinda)
Manually defining `HAVE_QHULL` before including pcl_config.h (happens in any PCL header), compilation succeeds, but linking fails:
```
// ConvexHull Test.cpp

#define HAVE_QHULL

#include "pcl/point_types.h"
#include "pcl/point_cloud.h"
#include "pcl/surface/convex_hull.h"

int main() {
    pcl::PointCloud<pcl::PointXYZ> pc;
    pcl::ConvexHull<pcl::PointXYZ> hull;

    hull.reconstruct(pc);
}
```
Build log:
```
1>ConvexHull Test.obj : error LNK2001: unresolved external symbol "protected: virtual void __cdecl pcl::ConvexHull<struct pcl::PointXYZ>::performReconstruction(struct pcl::PolygonMesh &)" (?performReconstruction@?$ConvexHull@UPointXYZ@pcl@@@pcl@@MEAAXAEAUPolygonMesh@2@@Z)
1>ConvexHull Test.obj : error LNK2001: unresolved external symbol "protected: virtual void __cdecl pcl::ConvexHull<struct pcl::PointXYZ>::performReconstruction(class std::vector<struct pcl::Vertices,class std::allocator<struct pcl::Vertices> > &)" (?performReconstruction@?$ConvexHull@UPointXYZ@pcl@@@pcl@@MEAAXAEAV?$vector@UVertices@pcl@@V?$allocator@UVertices@pcl@@@std@@@std@@@Z)
1>ConvexHull Test.obj : error LNK2001: unresolved external symbol "public: void __cdecl pcl::ConvexHull<struct pcl::PointXYZ>::reconstruct(class pcl::PointCloud<struct pcl::PointXYZ> &)" (?reconstruct@?$ConvexHull@UPointXYZ@pcl@@@pcl@@QEAAXAEAV?$PointCloud@UPointXYZ@pcl@@@2@@Z)
1>C:\ConvexHull Test\x64\Release\ConvexHull Test.exe : fatal error LNK1120: 3 unresolved externals
```

BTW, while you take a look, note that also the visualizer doesn't complie, maybe it's related.
```
include\pcl\visualization\point_cloud_geometry_handlers.h(49,10): fatal error C1083: Cannot open include file: 'vtkSmartPointer.h': No such file or directory (compiling source file ConvertPngToCloud.cpp)
```

--- a/ports/pcl/fix-find-qhull.patch
+++ b/ports/pcl/fix-find-qhull.patch
@@ -1,18 +1,8 @@
-From 7e3117fb47f58c7b1fb83e3a062a630b787a43bc Mon Sep 17 00:00:00 2001
-From: raahilsha-z <raahil.sha@zimaging.io>
-Date: Wed, 7 Jul 2021 16:11:12 -0400
-Subject: [PATCH] fix find qhull
-
----
- CMakeLists.txt         | 5 +----
- surface/CMakeLists.txt | 9 +++++++--
- 2 files changed, 8 insertions(+), 6 deletions(-)
-
 diff --git a/CMakeLists.txt b/CMakeLists.txt
-index a1d9bb58f..7cf86f74b 100644
+index a1d9bb5..73b1465 100644
 --- a/CMakeLists.txt
 +++ b/CMakeLists.txt
-@@ -359,10 +359,7 @@ if(WITH_QHULL)
+@@ -359,10 +359,8 @@ if(WITH_QHULL)
    if(NOT PCL_SHARED_LIBS OR ((WIN32 AND NOT MINGW) AND NOT PCL_BUILD_WITH_QHULL_DYNAMIC_LINKING_WIN32))
      set(QHULL_USE_STATIC ON)
    endif()
@@ -21,11 +11,12 @@ index a1d9bb58f..7cf86f74b 100644
 -    include_directories(SYSTEM ${QHULL_INCLUDE_DIRS})
 -  endif()
 +  find_package(Qhull CONFIG REQUIRED)
++  set(HAVE_QHULL ON)
  endif()
  
  # Cuda
 diff --git a/surface/CMakeLists.txt b/surface/CMakeLists.txt
-index d8a8566ea..1953c5566 100644
+index d8a8566..1953c55 100644
 --- a/surface/CMakeLists.txt
 +++ b/surface/CMakeLists.txt
 @@ -12,7 +12,7 @@ if(NOT build)
@@ -51,6 +42,3 @@ index d8a8566ea..1953c5566 100644
    target_link_libraries("${LIB_NAME}" ${QHULL_LIBRARIES})
  endif()
  
--- 
-2.32.0.windows.1
-

--- a/ports/pcl/portfile.cmake
+++ b/ports/pcl/portfile.cmake
@@ -42,14 +42,11 @@ vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
         cuda            BUILD_GPU
         tools           BUILD_tools
         opengl          WITH_OPENGL
-        vtk             WITH_VTK
         libusb          WITH_LIBUSB
+        visualization   WITH_VTK
         visualization   BUILD_visualization
         examples        BUILD_examples
         apps            BUILD_apps
-        apps            BUILD_apps_cloud_composer
-        apps            BUILD_apps_modeler
-        apps            BUILD_apps_point_cloud_editor
         # These 2 apps need openni1
         #apps            BUILD_apps_in_hand_scanner
         #apps            BUILD_apps_3d_rec_framework

--- a/ports/pcl/vcpkg.json
+++ b/ports/pcl/vcpkg.json
@@ -4,6 +4,7 @@
   "port-version": 4,
   "description": "Point Cloud Library (PCL) is open source library for 2D/3D image and point cloud processing.",
   "homepage": "https://github.com/PointCloudLibrary/pcl",
+  "license": "BSD-3-Clause",
   "supports": "!(arm64 & windows)",
   "dependencies": [
     "boost-asio",

--- a/ports/pcl/vcpkg.json
+++ b/ports/pcl/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "pcl",
   "version": "1.12.0",
-  "port-version": 3,
+  "port-version": 4,
   "description": "Point Cloud Library (PCL) is open source library for 2D/3D image and point cloud processing.",
   "homepage": "https://github.com/PointCloudLibrary/pcl",
   "supports": "!(arm64 & windows)",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -5238,7 +5238,7 @@
     },
     "pcl": {
       "baseline": "1.12.0",
-      "port-version": 3
+      "port-version": 4
     },
     "pcre": {
       "baseline": "8.45",

--- a/versions/p-/pcl.json
+++ b/versions/p-/pcl.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "3eb44f07f0155cf97aa0f3526606e0c7009466a3",
+      "version": "1.12.0",
+      "port-version": 4
+    },
+    {
       "git-tree": "0f0c24ef83692fbcc9434ec9622592ba9b16e0d5",
       "version": "1.12.0",
       "port-version": 3

--- a/versions/p-/pcl.json
+++ b/versions/p-/pcl.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "3eb44f07f0155cf97aa0f3526606e0c7009466a3",
+      "git-tree": "7e54ab86574dfc901275996e282f8db5c6c9c3fb",
       "version": "1.12.0",
       "port-version": 4
     },

